bftengine/src/bcstatetransfer/BCStateTran.hpp

```diff
@@ -301,10 +303,16 @@ class BCStateTran : public IStateTransfer {
   static const uint64_t ID_OF_VBLOCK_RES_PAGES = UINT64_MAX;
 
   uint64_t nextRequiredBlock_ = 0;
-  uint64_t nextCommittedBlockId_ = 0;
+  uint64_t nextBlockIdToCommit_ = 0;
+  DataStore::CheckpointDesc targetCheckpointDesc_;
   STDigest digestOfNextRequiredBlock_;
   bool posponedSendFetchBlocksMsg_;
 
+  inline bool isMinBlockIdInCurrentRange(uint64_t blockId) const;
+  inline bool isMaxBlockIdInCurrentRange(uint64_t blockId) const;
```

Comment:
Should all Block operations be put at single place? Maybe some new class BlockOps?

Reply:
I think the ordering of the functions in the class is important, I agree.
But if we do it now, we will have to do it again later so we do double work.
I would 1st have the code done an then refactor - see the big picture - refactor to files and classes, set visibility etc.
Now it will be very partial work.
When I put it here I knew it will change again soon.